gripmock.go

```diff
@@ -95,18 +96,45 @@ type protocParam struct {
 	imports     []string
 }
 
-func generateProtoc(param protocParam) {
-	param.protoPath = fixGoPackage(param.protoPath)
-	protodirs := strings.Split(param.protoPath[0], "/")
+func getProtodirs(protoPath string, imports []string) []string {
+	// deduced protodir from protoPath
+	splitpath := strings.Split(protoPath, "/")
 	protodir := ""
-	if len(protodirs) > 0 {
-		protodir = strings.Join(protodirs[:len(protodirs)-1], "/") + "/"
+	if len(splitpath) > 0 {
+		protodir = path.Join(splitpath[:len(splitpath)-1]...)
 	}
 
-	args := []string{"-I", protodir}
+	// search protodir prefix
+	protodirIdx := -1
+	for i := range imports {
+		dir := path.Join("protogen", imports[i])
+		if strings.HasPrefix(protodir, dir) {
+			protodir = dir
+			protodirIdx = i
+			break
+		}
+	}
+
+	protodirs := make([]string, 0, len(imports)+1)
+	protodirs = append(protodirs, protodir)
 	// include well-known-types
-	for _, i := range param.imports {
-		args = append(args, "-I", i)
+	for i, dir := range imports {
```

Comment:
the comment above it is obsolete, please update it to explain what happened

Reply:
updated